misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/options/FormattingOptions.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.formatter.core.options;
+
+import org.ballerinalang.formatter.core.FormatterException;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+import static org.ballerinalang.formatter.core.FormatterUtils.getFormattingConfigurations;
+
+/**
+ * A model for formatting options that could be passed onto the formatter.
+ */
+public class FormattingOptions {
+
+    private IndentFormattingOptions indentFormattingOptions;
+    private WrappingFormattingOptions wrappingFormattingOptions;
+    private BraceFormattingOptions braceFormattingOptions;
+    private FunctionDeclFormattingOptions functionDeclFormattingOptions;
+    private FunctionCallFormattingOptions functionCallFormattingOptions;
+    private IfStatementFormattingOptions ifStatementFormattingOptions;
+    private SpacingFormattingOptions spacingFormattingOptions;
+    private ForceFormattingOptions forceFormattingOptions;
+    private ImportFormattingOptions importFormattingOptions;
+    private QueryFormattingOptions queryFormattingOptions;
```

Comment:
Is there a reason why these are not final? Cannot seem to find instances where you modify them.